core/src/main/scala/org/http4s/HttpApp.scala

```diff
@@ -21,7 +21,7 @@ object HttpApp {
     * @param run the function to lift
     * @return an [[HttpApp]] that wraps `run`
     */
-  def apply[F[_]: Sync](run: Request[F] => F[Response[F]]): HttpApp[F] =
+  def apply[F[_]: Defer](run: Request[F] => F[Response[F]]): HttpApp[F] =
```

Comment:
Prior discussion on https://github.com/http4s/http4s/pull/2325. @SystemFw and @kubukoz may have evolved their thinking in a cats-effect-3 world.

Reply:
I guess we already did this on `Http`: https://github.com/http4s/http4s/pull/2906

@ChristopherDavenport, are you on board with the rest of this, then?